libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/FreeDriveCarScreen.kt

```diff
@@ -13,57 +13,60 @@ import com.mapbox.androidauto.car.navigation.CarNavigationCamera
 import com.mapbox.androidauto.car.navigation.roadlabel.RoadLabelSurfaceLayer
 import com.mapbox.androidauto.car.navigation.speedlimit.CarSpeedLimitRenderer
 import com.mapbox.androidauto.car.preview.CarRouteLine
+import com.mapbox.androidauto.internal.car.extensions.addBackPressedHandler
 import com.mapbox.androidauto.internal.logAndroidAuto
 import com.mapbox.maps.MapboxExperimental
 
 /**
  * When the app is launched from Android Auto
  */
 @OptIn(MapboxExperimental::class)
-class MainCarScreen @UiThread constructor(
-    private val mainCarContext: MainCarContext
-) : Screen(mainCarContext.carContext) {
+class FreeDriveCarScreen @UiThread constructor(
+    private val mapboxCarContext: MapboxCarContext
+) : Screen(mapboxCarContext.carContext) {
 
     val carRouteLine = CarRouteLine()
     val carLocationRenderer = CarLocationRenderer()
-    val carSpeedLimitRenderer = CarSpeedLimitRenderer(mainCarContext)
+    val carSpeedLimitRenderer = CarSpeedLimitRenderer(mapboxCarContext)
     val carNavigationCamera = CarNavigationCamera(
         initialCarCameraMode = CarCameraMode.FOLLOWING,
         alternativeCarCameraMode = null,
     )
     private val roadLabelSurfaceLayer = RoadLabelSurfaceLayer(carContext)
-    private val mainActionStrip = MainActionStrip(this, mainCarContext)
     private val mapActionStripBuilder = MainMapActionStrip(this, carNavigationCamera)
 
     init {
+        addBackPressedHandler {
+            mapboxCarContext.mapboxScreenManager.goBack()
+        }
```

Comment:
Do we need this? `FreeDriveCarScreen` doesn't have a back button.

Reply:
Thanks! it's not needed.

thought about the possibility of adding `Action.BACK` and `addBackPressedHandler` programmatically when there exists a backstack. maybe if we make a "MapboxScreenBuilder". not today